From fde3f8fe2088c8acc5cc71b3479656a76c5615fc Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 20 Jul 2017 10:19:43 +0800
Subject: [PATCH 2176/5242] MLK-16036 drm/imx: ldb: Avoid early return when
 getting aux PHY in dual mode

commit  948da630e4a927366017d7ab7e1ee680d2f9e270 from
https://source.codeaurora.org/external/imx/linux-imx.git

In dual mode, we return too early from ->bind when we get
the auxiliary channel's PHY.  This causes we miss the logics
to set driver data, get ldb alias id and initialize pixel
link(if necessary).  This patch fixes the issue here by
tweaking the driver logic to do component binding properly.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/imx-ldb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/imx-ldb.c b/drivers/gpu/drm/imx/imx-ldb.c
index 7ed0642..3f7b474 100644
--- a/drivers/gpu/drm/imx/imx-ldb.c
+++ b/drivers/gpu/drm/imx/imx-ldb.c
@@ -1224,7 +1224,7 @@ static int imx_ldb_bind(struct device *dev, struct device *master, void *data)
 			}
 
 			if (auxiliary_ch)
-				return 0;
+				continue;
 		}
 
 		ret = imx_ldb_register(drm, channel);
-- 
1.7.9.5

